provider/testing/resource.go

```diff
@@ -126,35 +131,70 @@ func deleteTables(conn *pgxpool.Conn, table *schema.Table) error {
 }
 
 func verifyNoEmptyColumns(t *testing.T, tc ResourceTestCase, conn pgxscan.Querier) {
+	t.Helper()
+	if tc.SkipEmptyRows {
+		t.Logf("table %s marked with SkipEmptyRows. Skipping...", tc.Table.Name)
+		return
+	}
 	// Test that we don't have missing columns and have exactly one entry for each table
 	for _, table := range getTablesFromMainTable(tc.Table) {
-		query := fmt.Sprintf("select * FROM %s ", table)
-		rows, err := conn.Query(context.Background(), query)
+		if table.IgnoreInTests {
+			t.Logf("table %s marked as IgnoreInTest. Skipping...", table.Name)
+			continue
+		}
+		s := sq.StatementBuilder.
+			PlaceholderFormat(sq.Dollar).
+			Select(fmt.Sprintf("json_agg(%s)", table.Name)).
+			From(table.Name)
+		query, args, err := s.ToSql()
 		if err != nil {
 			t.Fatal(err)
 		}
-		count := 0
-		for rows.Next() {
-			count += 1
+		var data []map[string]interface{}
+		if err := pgxscan.Get(context.Background(), conn, &data, query, args...); err != nil {
+			t.Fatal(err)
 		}
-		if count < 1 {
-			t.Fatalf("expected to have at least 1 entry at table %s got %d", table, count)
+
+		if len(data) == 0 {
+			t.Errorf("expected to have at least 1 entry at table %s got zero", table.Name)
+			return
 		}
-		if tc.SkipEmptyJsonB {
-			continue
+
+		nilColumns := map[string]bool{}
+		// mark all columns as nil
+		for _, c := range table.Columns {
+			if !c.IgnoreInTests {
+				nilColumns[c.Name] = true
+			}
 		}
-		query = fmt.Sprintf("select t.* FROM %s as t WHERE to_jsonb(t) = jsonb_strip_nulls(to_jsonb(t))", table)
-		rows, err = conn.Query(context.Background(), query)
-		if err != nil {
-			t.Fatal(err)
+
+		for _, row := range data {
+			for c, v := range row {
+				if v != nil {
+					// as long as we had one row or result with this column not nil it means the resolver worked
+					nilColumns[c] = false
+				}
+			}
 		}
-		count = 0
-		for rows.Next() {
-			count += 1
+
+		var nilColumnsArr []string
+		for c, v := range nilColumns {
+			if v {
+				nilColumnsArr = append(nilColumnsArr, c)
+			}
 		}
-		if count < 1 {
-			t.Fatalf("row at table %s has an empty column", table)
+
+		if len(nilColumnsArr) != 0 {
+			b, err := json.MarshalIndent(data, "", "\t")
+			if err != nil {
+				t.Fatal(err)
+			}
+			t.Errorf("found nil column in table %s. rows=\n%s\ncolumns=%s\n", table.Name, string(b), strings.Join(nilColumnsArr, ","))
 		}
+		// if tc.SkipEmptyJsonB {
+		// 	continue
+		// }
```

Comment:
remove?

Reply:
Im not sure where it is used. I don't think it is used in aws at least. if it's used somewhere else we can put it back.